src/opencl/utils.rs

```diff
@@ -1,19 +1,45 @@
 use std::convert::{TryFrom, TryInto};
-use std::fmt::Write;
 
 use lazy_static::lazy_static;
 use log::{debug, warn};
 use opencl3::device::DeviceInfo::CL_DEVICE_GLOBAL_MEM_SIZE;
 use sha2::{Digest, Sha256};
 
-use super::{Device, GPUError, GPUResult, Vendor};
+use super::{Device, DeviceUuid, GPUError, GPUResult, PciId, Vendor, CL_UUID_SIZE_KHR};
 
-fn get_bus_id(d: &opencl3::device::Device) -> Result<u32, GPUError> {
-    let vendor = Vendor::try_from(d.vendor()?.as_str())?;
-    match vendor {
-        Vendor::Amd => d.pci_bus_id_amd().map_err(Into::into),
-        Vendor::Nvidia => d.pci_bus_id_nv().map_err(Into::into),
-    }
+/// The PCI-ID is the combination of the PCI Bus ID and PCI Device ID.
+///
+/// It is the first two identifiers of e.g. `lcpci`:
```

Comment:
typo

Reply:
There were more places then I thought with this typo ;)